packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -288,6 +300,79 @@ class FuchsiaDevice extends Device {
     throwToolExit('No ports found running $isolateName');
     return null;
   }
+
+  FuchsiaIsolateDiscoveryProtocol  getIsolateDiscoveryProtocol(String isolateName) => FuchsiaIsolateDiscoveryProtocol(this, isolateName);
+}
+
+class FuchsiaIsolateDiscoveryProtocol {
+  FuchsiaIsolateDiscoveryProtocol(this._device, this._isolateName);
+
+  static const Duration _pollDuration = Duration(seconds: 10);
+  final Map<int, VMService> _ports = <int, VMService>{};
+  final FuchsiaDevice _device;
+  final String _isolateName;
+  final Completer<Uri> _foundUri = Completer<Uri>();
+  Timer _pollingTimer;
+  Status _status;
+  bool _locked = false;
+
+  Future<Uri> get uri {
```

Comment:
Maybe to avoid unnecessary awaits:
```dart
Uri _uri;
FutureOr<Uri> get uri {
  if (_uri != null) return _uri;
  ...
  return _foundUri.future.then((uri) {
    _uri = uri;
    return uri;
  });
}
```

Reply:
Done